File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        zero_at = -1
        pdct = 1
        ret = [0]*len(nums)
        for i in range(0,len(nums)):
            if nums[i] == 0:
                if zero_at > -1:
                    return ret
                zero_at = i
            else:
                pdct*=nums[i]
            
        if zero_at > -1:
            ret[zero_at] = pdct
        else:
            for i in range(0,len(nums)):
                ret[i]=int(pdct/nums[i])
        return ret
        
            

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int i, j;
        i=j=1;
        while(i<nums.size()) {
            while(nums[i] == nums[i-1]) i++;
            if(i<nums.size()) {
                nums[j] = nums[i];
                j++;
                i++;
            }
        }
        return j;
    }
};